lib/ansible/modules/cloud/amazon/glue_job.py

```
@@ -0,0 +1,377 @@
+    if module.params.get("command_script_location") is not None:
+        params['Command'] = {'Name': module.params.get("command_name"), 'ScriptLocation': module.params.get("command_script_location")}
+    if module.params.get("connections") is not None:
+        params['Connections'] = {'Connections': module.params.get("connections")}
+    if module.params.get("default_arguments") is not None:
+        params['DefaultArguments'] = module.params.get("default_arguments")
+    if module.params.get("description") is not None:
+        params['Description'] = module.params.get("description")
+    if module.params.get("max_concurrent_runs") is not None:
+        params['ExecutionProperty'] = {'MaxConcurrentRuns': module.params.get("max_concurrent_runs")}
+    if module.params.get("max_retries") is not None:
+        params['MaxRetries'] = module.params.get("max_retries")
+    if module.params.get("timeout") is not None:
+        params['Timeout'] = module.params.get("timeout")
+
+    # If glue_job is not None then check if it needs to be modified, else create it
+    if glue_job:
+        if _compare_glue_job_params(params, glue_job):
+            try:
+                # Update job needs slightly modified params
+                update_params = {'JobName': params['Name'], 'JobUpdate': copy.deepcopy(params)}
+                del update_params['JobUpdate']['Name']
+                connection.update_job(**update_params)
+                changed = True
+            except (BotoCoreError, ClientError) as e:
+                module.fail_json_aws(e)
+    else:
+        try:
+            connection.create_job(**params)
+            changed = True
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e)
+
+    # If changed, get the Glue job again
+    if changed:
+        glue_job = _get_glue_job(connection, module, params['Name'])
+
+    module.exit_json(changed=changed, **camel_dict_to_snake_dict(glue_job))
```

We've been having issues in other tests (`ec2_vpc_*`, `ec2_group`, etc) caused by AWS API's not being read-back consistent. I see the `_get_glue_job` call will silently return `None` if the job isn't found. Since at :312 and here you know the job exists, can you add a check that will retry if the job isn't found? I think camel2snake raises an exception if it's given a `None` as the parameter.

---

Is there a specific way you'd like to implement that or just an `while glue_job is None` loop will do?

---

Yeah, a simple loop like that would be perfect - I just want to protect against accidentally getting `None` from `_get_glue_job` because the AWS API hasn't seen the change yet.